File: containsDuplicates.py
def count(list):
    r = 0
    for x in list:
        r = r + 1;

    return r;

class Solution:
    
    def containsDuplicate(self, nums: List[int]) -> bool:
        nums_no_duplicates = list(dict.fromkeys(nums))
        
        og_count = count(nums)
        filtered_count = count(nums_no_duplicates)
        
        return og_count > filtered_count
      
      
# Runtime: 521 ms, faster than 77.44% of Python3 online submissions for Contains Duplicate.
# Memory Usage: 25.9 MB, less than 72.31% of Python3 online submissions for Contains Duplicate.

File: containsDuplicates.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int ogNumsCount = 0;
        for (int x: nums) {
            ogNumsCount++;
        }
        
        sort(nums.begin(), nums.end());
        nums.erase(unique(nums.begin(), nums.end() ), nums.end());   
        
        int uniqueNumsCount = 0;
        for (int x: nums) {
            uniqueNumsCount++;
        }
        
        return ogNumsCount > uniqueNumsCount;
    }
};

// Runtime: 114 ms, faster than 79.25% of C++ online submissions for Contains Duplicate.
// Memory Usage: 46.5 MB, less than 97.54% of C++ online submissions for Contains Duplicate.
